src/dist/replication/meta_server/duplication/meta_duplication_service.cpp

```diff
@@ -231,5 +231,106 @@ meta_duplication_service::new_dup_from_init(const std::string &remote_cluster_na
     return dup;
 }
 
+// ThreadPool(WRITE): THREAD_POOL_META_STATE
+void meta_duplication_service::recover_from_meta_state()
+{
+    ddebug_f("recovering duplication states from meta storage");
+
+    // /<app>/duplication/<dupid>/<partition_idx>
+    //                       |         |-> confirmed_decree
+    //                       |
+    //                       |-> json of dup info
+
+    for (const auto &kv : _state->_exist_apps) {
+        std::shared_ptr<app_state> app = kv.second;
+        if (app->status != app_status::AS_AVAILABLE) {
+            return;
+        }
+
+        _meta_svc->get_meta_storage()->get_children(
+            get_duplication_path(*app),
+            [this, app](bool node_exists, const std::vector<std::string> &dup_id_list) {
+                if (!node_exists) {
+                    // if there's no duplication
```

Comment:
I suggest printing log here, ddebug or dinfo

Reply:
Should perform no additional behavior when the app does not have duplication.

Reply:
I recommend add log here and return directly